### PR TITLE
[14.0][MIG] sale_mrp_bom: Migration to 14.0

### PR DESCRIPTION
**Pull request by hailangvn**

### CODE REVIEW

**Review by StephaneMangin — APPROVED** [review]
LGTM

**Review by i-vyshnevska — COMMENTED** [review]
seems like git history is lost
code looks LG

**Comment by sebalix** [comment]
Git history seems good (=> https://github.com/OCA/sale-workflow/commits/12.0/sale_mrp_bom)

**Review by sebalix — APPROVED** [review]

File: sale_mrp_bom/models/sale_order_line.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+from odoo import fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    bom_id = fields.Many2one(
+        comodel_name="mrp.bom",
+        string="BOM",
+        domain="[('product_tmpl_id.product_variant_ids', '=', product_id),"
```
Comment (CasVissers-360ERP) [inline sale_mrp_bom/models/sale_order_line.py:14]: Is this domain sufficient? When creating a new BoM the product of the new BoM can be different to the product of the sale order line? Maybe add a constraint for this?

File: sale_mrp_bom/views/sale_order.xml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    <record id="sale_order_form" model="ir.ui.view">
+        <field name="name">sale_mrp_bom.sale.order.form</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form" />
+        <field name="arch" type="xml">
+            <xpath
+                expr="//field[@name='order_line']//tree//field[@name='name']"
+                position="after"
+            >
+                <field name="bom_id" groups="sale_mrp_bom.sale_mrp_bom_group" />
```
Comment (CasVissers-360ERP) [inline sale_mrp_bom/views/sale_order.xml]: Add  context="{'default_product_id': product_id, 'default_product_tmpl_id': product_template_id}"

File: sale_mrp_bom/views/sale_order.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+            </xpath>
+            <xpath
+                expr="//field[@name='order_line']//form//field[@name='customer_lead']"
+                position="after"
+            >
+                <field name="bom_id" groups="sale_mrp_bom.sale_mrp_bom_group" />
```
Comment (CasVissers-360ERP) [inline sale_mrp_bom/views/sale_order.xml]: Add  context="{'default_product_id': product_id, 'default_product_tmpl_id': product_template_id}"

File: sale_mrp_bom/models/sale_order_line.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+from odoo import fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    bom_id = fields.Many2one(
+        comodel_name="mrp.bom",
+        string="BOM",
```
Comment (CasVissers-360ERP) [inline sale_mrp_bom/models/sale_order_line.py]: Odoo naming is BoM not BOM

**Review by CasVissers-360ERP — CHANGES_REQUESTED** [review]
Functional review

**Comment by rousseldenis** [comment]
@hailangvn  Could you attend @CasVissers-360ERP comments ?

**Review by CasVissers-360ERP — APPROVED** [review]
LGTM, thanks for the improvements!

**Comment by hailangvn** [comment]
Good day @pedrobaeza, I am first time contributor so this PR need manual approval. Could you please help or advise what to do? Thank you in advance.

**Comment by pedrobaeza** [comment]
Approved.

**Pull request merged**